### PR TITLE
Validate against past dates in create posting form

### LINKED ISSUE #283: Dates occurring in the past is not flagged as an error
### Description

Posting Closing Date, Start Date, and End Date fields can be set to the past.

### Acceptance Criteria

Super admin cannot set these fields to dates in the past (preferred).

OR

An error appears when these dates are set to dates in the past.

### Technical
In `frontend\src\components\admin\posting\CreatePostingShifts.tsx` in the `handleNext()` method, we are currently checking `if (startDate > endDate)` . We also want to check `startDate < today` or `endDate < today` .  If so, set the startDateError / setEndDateError .

### PR DESCRIPTION
**Pull request by sherryhli**
## Ticket link

Closes #283 

## Implementation description
* Changed error state boolean variables to strings that contains the error message
* Validate that selected dates are not in the past in the `handleNext()` functions

## Steps to test
1. Use the preview deploy link to test if the dates in past case is being correctly validated

## What should reviewers focus on?
General code correctness.

**Known caveat:** the new and longer error messages are surfacing a layout bug. Unfortunately I don't have the capacity to address this right now.
Capture2
Capture

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by LenaNguyen** [comment]
For the Posting Closing date, I chose a date in the future but i still get the error message

File: frontend/src/components/admin/posting/CreatePostingShifts.tsx [hunk 1]
```diff
@@ -152,17 +161,53 @@ const CreatePostingShifts: React.FC<CreatePostingShiftsProps> = ({
...
+        "Please ensure that end date is after start date.",
+      );
+    }
+
+    const now = new Date();
+    let currentMonth = `${now.getMonth() + 1}`;
+    if (currentMonth.length < 2) {
+      currentMonth = `0${currentMonth}`;
+    }
+    const currentDateISOString = `${now.getFullYear()}-${currentMonth}-${now.getDate()}`;
```
Comment (sherryhli) [inline frontend/src/components/admin/posting/CreatePostingShifts.tsx:189]: Should add additional logic to format the day part of the date to 2 digits always

**Comment by sherryhli** [comment]
> Just the comment I left. Does the posting closing date have to be a certain number of days in the future? If so, we should make the error message clearer

Sorry, this is a bug! See comment I just made

File: frontend/src/components/admin/posting/CreatePostingBasicInfo.tsx [hunk 2]
```diff
@@ -269,22 +272,38 @@ const CreatePostingBasicInfo: React.FC<CreatePostingBasicInfoProps> = ({
...
     setSkillsError(selectedSkills.length < 1);
     setEmployeesError(selectedEmployees.length < 1);
 
-    if (
+    const now = new Date();
+    let currentMonth = `${now.getMonth() + 1}`;
+    if (currentMonth.length < 2) {
+      currentMonth = `0${currentMonth}`;
+    }
+    const currentDateISOString = `${now.getFullYear()}-${currentMonth}-${now.getDate()}`;
```
Comment (sherryhli) [inline frontend/src/components/admin/posting/CreatePostingBasicInfo.tsx:289]:
```suggestion
    let currentDay = `${now.getDate()}`;
    if (currentDay.length < 2) {
        currentDay = `0${currentDay}`;
    }
    const currentDateISOString = `${now.getFullYear()}-${currentMonth}-${currentDay}`;
```

File: frontend/src/components/admin/posting/CreatePostingShifts.tsx (hunk 1 above)
Comment (sherryhli) [inline frontend/src/components/admin/posting/CreatePostingShifts.tsx:189]:
```suggestion
    let currentDay = `${now.getDate()}`;
    if (currentDay.length < 2) {
        currentDay = `0${currentDay}`;
    }
    const currentDateISOString = `${now.getFullYear()}-${currentMonth}-${currentDay}`;
```

**Review by LenaNguyen — APPROVED** [review]

**Pull request merged**